src/sync/patch.rs

```diff
@@ -1,66 +1,50 @@
 use crate::db;
 use crate::util::rlog;
 use serde::Deserialize;
-use serde_json::json;
-use std::default::Default;
 
-const OP_ADD: &str = "add";
-const OP_REMOVE: &str = "remove";
-const OP_REPLACE: &str = "replace";
-
-#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
-pub struct Operation {
-    pub op: String,
-    pub path: String,
-    #[serde(default)]
-    pub value: serde_json::Value,
+#[derive(Deserialize)]
+#[cfg_attr(test, derive(Clone, Debug, PartialEq))]
+#[serde(tag = "op")]
```

Comment:
This worked out nicely. I was half expecting to have to write my own Deserializer.

Reply:
yeah that is really nice, TIL